aQute.libg/src/aQute/libg/pack200/Unpack200.java

```diff
@@ -0,0 +1,147 @@
+package aQute.libg.pack200;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import aQute.libg.command.Command;
+
+public class Unpack200 {
+
+	public static final String	PACKED_SUFFIX		= ".pack.gz";
+	public static final String	JAR_SUFFIX			= ".jar";
+
+	public static final String	PACK200_PROPERTY	= "aQute.bnd.pack200";
+	public static final String	JRE					= "@jre";
+	public static final String	PATH				= "@path";
+	public static final String	NONE				= "@none";
+
+	private static Boolean	canUnpack		= null;
+	private static String	unpackCommand	= null;
+
+	private final static Logger	logger		= LoggerFactory.getLogger(Unpack200.class);
+
+	static {
+		canUnpack();
+	}
+
+	private Unpack200() {}
+
+	public static boolean canUnpack() {
+		if (canUnpack != null)
+			return canUnpack.booleanValue();
+
+		String[] locations = getUnpack200Commands();
+		if (locations == null) {
+			canUnpack = Boolean.FALSE;
+			unpackCommand = null;
+			return false;
+		}
+
+		int result;
+		for (String location : locations) {
+			if (location == null) {
+				continue;
+			}
+			StringBuffer sb = new StringBuffer();
+			Command cmd = new Command();
+			cmd.add(location);
+			cmd.add("-V");
+			try {
+				logger.debug("Calling: {}", cmd.toString());
+				result = cmd.execute(sb, sb);
+			} catch (Exception e) {
+				logger.error("Error: " + sb.toString(), e);
+				result = -1;
+			}
+			if (result == 0) {
+				unpackCommand = location;
+				canUnpack = Boolean.TRUE;
+				return true;
+			}
+		}
+		canUnpack = Boolean.FALSE;
+		return false;
+	}
+
+	public static File unpack(File input) {
+		File workingDirectory = input.getParentFile();
+		String name = input.getName();
+		if (name.endsWith(PACKED_SUFFIX)) {
+			name = name.substring(0, name.length() - PACKED_SUFFIX.length());
+		}
+		if (!name.endsWith(JAR_SUFFIX)) {
+			name = name + JAR_SUFFIX;
+		}
+		File output = new File(workingDirectory, name);
+		return unpack(input, output);
+	}
+
+	public static File unpack(File input, File output) {
+		if (!canUnpack()) {
+			throw new IllegalStateException("unpack200 not found on this system");
+		}
+		File parent = output.getParentFile();
+		if (!parent.exists())
+			parent.mkdirs();
+
+		StringBuffer sb = new StringBuffer();
+		try {
+
+			Command cmd = new Command();
+			cmd.add(unpackCommand);
+			cmd.add(input.getCanonicalPath());
+			cmd.add(output.getCanonicalPath());
+
+			logger.debug("Calling: {}", cmd.toString());
+			int result = cmd.execute(sb, sb);
+
+			return output;
+		} catch (IOException e) {
+			logger.error(e.getMessage(), e);
+			throw new RuntimeException(e);
+		} catch (Exception e) {
+			logger.error("Error: " + sb.toString(), e);
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Get unpack200 command
+	 *
+	 * @return String [] or null
+	 */
+	public static String[] getUnpack200Commands() {
```

Comment:
Once this class is in biz.aQute.repository, we can use Processor.getJavaExecutable to find the unpack command. See

https://github.com/bndtools/bnd/blob/98e97befcd426ecd99bc85ed32f421bf95c98bdf/biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Tool.java#L160-L161

where we use this method to find the javadoc command.

This method lets the user set the `unpack200` property in their workspace or project to override the default command. Then we don't need `@jre`, `@path`, `@none`. We just use the value of the `unpack200` property (like we do for the `javadoc`, `java`, and `javac` properties) to find the command.